buffer.go

```diff
@@ -33,6 +33,37 @@ func SingleTagBuffer(key string, value interface{}) *Buffer {
 	return b
 }
 
+// BuildBuffer is used to build a *Buffer that contains an arbitrary number
+// of tags. Sample usage:
+//
+//	bld := BuildBuffer()
+//	bld.Add("a", 1)
+//	bld.Add("b", 2)
+//	buf := bld.Finish()
+func BuildBuffer() BufferBuilder {
+	bld := BufferBuilder{b: &Buffer{}}
+	bld.b.init(0, 0)
+	return bld
+}
+
+// BufferBuilder is returned by BuildBuffer.
+type BufferBuilder struct {
+	b *Buffer
+}
+
+// Add a log tag to the buffer. If the key was added already, the value is
+// replaced.
+func (bld BufferBuilder) Add(key string, value interface{}) {
```

Comment:
Assuming these are switched to pointer receivers (see below), any interest in returning the receiver so that these calls can be chained?

Reply:
I started that way but separating the calls looks cleaner IMO  BTW could there be an issue with the pointer escaping if we return it? We definitely want the builder itself to stay on stack.

Reply:
I was wondering the same. This specific method will probably be mid-stack inlined, so it shouldn't matter, but I'm not sure if a non-inlined method which returns the receiver always causes that receiver to escape.

If you don't think chaining is useful, then we can just skip it to avoid the question.